packages/grid/src/mouse-handlers/GridLinkMouseHandler.ts

```diff
@@ -0,0 +1,256 @@
+
+    const {
+      actualBoundingBoxAscent,
+      actualBoundingBoxDescent,
+    } = context.measureText(truncatedText);
+    const linkHeight = actualBoundingBoxAscent + actualBoundingBoxDescent;
+
+    // Consider edge cases
+    const top = Math.max(gridY, gridY + textY - linkHeight / 2);
+    const bottom = clamp(
+      top + linkHeight,
+      top,
+      gridHeight - horizontalBarHeight
+    );
+
+    // Loop through the blocks of text
+    for (let i = 0; i < links.length; i += 1) {
+      const { end, start, href } = links[i];
+      let { value } = links[i];
+
+      if (end > truncatedText.length) {
+        value = truncatedText.substring(start);
+      }
+
+      let { width: linkWidth } = context.measureText(value);
+      // Measure the width of the substring before the link
+      const startX =
+        context.measureText(truncatedText.substring(0, start)).width + textX;
+      // Right side is greater than gridX
+
+      // Check if the x position is less than the grid x, then linkWidth should be shifted by gridX - startX
+      linkWidth -= startX < gridX ? gridX - startX : 0;
+
+      const left = Math.max(gridX, gridX + startX);
+      const right = clamp(left + linkWidth, left, gridWidth - verticalBarWidth);
+
+      const cursorIsInBounds =
+        x >= left && x <= right && y >= top && y <= bottom;
+
+      if (cursorIsInBounds) {
+        // Reset font
+        context.restore();
+
+        this.currentLinkBox = {
+          left,
+          top,
+          right,
+          bottom,
+          width: linkWidth,
+          height: linkHeight,
+          value,
+          href,
+        };
+
+        return this.currentLinkBox;
+      }
+    }
+
+    // If it reaches this point, then the mouse is not hovering a link so restore context and currentLinkBox
+    context.restore();
+    this.currentLinkBox = undefined;
+
+    return null;
+  }
+
+  /**
+   * Set the cursor to pointer when hovering of a link
+   * @param gridPoint Mouse position information
+   * @param grid The grid
+   * @returns False
+   */
+  private setCursor(gridPoint: GridPoint, grid: Grid): EventHandlerResult {
+    if (GridLinkMouseHandler.getHoveringLinkInfo(gridPoint, grid) != null) {
+      this.cursor = 'pointer';
+    } else {
+      this.cursor = null;
+    }
+    return false;
+  }
+
+  onMove(gridPoint: GridPoint, grid: Grid): EventHandlerResult {
+    return this.setCursor(gridPoint, grid);
+  }
+
+  onDown(
+    gridPoint: GridPoint,
+    grid: Grid,
+    event: GridMouseEvent
+  ): EventHandlerResult {
+    this.isDown = true;
+    const link = GridLinkMouseHandler.getHoveringLinkInfo(gridPoint, grid);
+
+    // If a modifier key or shift is down, we don't want to handle it and set isDown to false so onUp won't handle it either
+    if (GridUtils.isModifierKeyDown(event) || event.shiftKey || link == null) {
+      this.isDown = false;
+      return false;
+    }
+
+    if (this.timeoutId) {
+      clearTimeout(this.timeoutId);
+    }
```

Comment:
You don't need to surround with `if`, `clearTimeout` does nothing if you pass in `undefined`:
```suggestion
    clearTimeout(this.timeoutId);
```
(Also the `if` check wouldn't work correctly if `timeoutId` was `0`, since that's a falsey value. It's unlikely `timeoutId` is `0`, but I don't think impossible...)

Reply:
TypeScript complains about it being undefined.